### PR TITLE
Add synapse_densities.py tool, provide volume description

### PR DESCRIPTION
**Pull request by tribal-tec**

### CODE REVIEW

File: apps/synapseDensities/synapse_densities.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+                        help="fraction of GIDs to use [0,1]")
+    parser.add_argument("-d", "--datarange", nargs=2, metavar=('min', 'max'),
+                        help="data range to use")
+    parser.add_argument("-r", "--resolution", metavar="<micrometer/voxel>", default=16, type=float,
+                        help="resolution in micrometer/voxel, default 16")
+    parser.add_argument("--reference", metavar='<path to volume>', default="",
+                        help="path to reference volume for size and resolution setup")
+    parser.add_argument("-v", "--verbose", action="store_true",
+                        help="Print volume URI")
+    args = parser.parse_args()
```
Comment (eile) [inline apps/synapseDensities/synapse_densities.py]: an inverse urihandler! ;)

**Comment by tribal-tec** [comment]
retest this please

File: fivox/livre/dataSource.h [hunk 2]
```diff
@@ -54,7 +54,7 @@ class DataSource : public livre::DataSourcePlugin
      * which means that the number of frames changes over time.
      * This function updates the data source to the latest data received.
      */
-    void update() final;
+    bool update() final;
```
Comment (jafyvilla) [inline fivox/livre/dataSource.h]: doxygen return

File: fivox/uriHandler.cpp [hunk 3]
```diff
@@ -229,6 +229,51 @@ class URIHandler::Impl
...
+                desc << "compartment ";
+            desc << "report '" << getReport() << "' for target '"
+                 << _get( "target" ) << "'";
+            break;
+        case VolumeType::spikes:
+            desc << "number of spikes in " << getDuration()
+                 << "ms for target '" << _get( "target" ) << "'";
+            break;
+        case VolumeType::synapses:
+        {
```
Comment (jafyvilla) [inline fivox/uriHandler.cpp]: are these braces needed?

Comment (tribal-tec) [inline fivox/uriHandler.cpp]: :facepunch:

**Pull request merged**